src/FluentAssertions.Analyzers/Tips/Collections/CollectionShouldHaveElementAt.cs

```diff
@@ -27,6 +28,16 @@ protected override IEnumerable<FluentAssertionsCSharpSyntaxVisitor> Visitors
             }
         }
 
+        protected override bool ShouldAnalyzeVariableType(ITypeSymbol type)
+        {
+            if (type.AllInterfaces.Any(@interface => @interface.Name == "IReadOnlyDictionary" || @interface.Name == "IDictionary"))
```

Comment:
Since the dictionary assertions are on the type `IDictionary` I think this should not look for the readonly one

Reply:
That's correct, however, this is preventing the HaveIndexAt one from incorrectly firing on dictionaries. Since IDictionary doesn't inherit IReadOnlyDictionary I've done two checks (I use IReadOnlyDictionary in my interfaces which is where I found the bug).